client/client.go

```diff
@@ -349,11 +349,14 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 
 	select {
 	case <-ctx.Done():
-		err = resp.Body.Close()
-		<-done
-		if err == nil {
-			err = ctx.Err()
+		if err = resp.Body.Close(); err != nil {
```

Comment:
is this correct? breaking might cause the reading routine to be blocking forever. Why do we want to break out in this case?